core/src/main/java/org/elasticsearch/action/admin/indices/mapping/put/PutMappingRequest.java

```diff
@@ -98,11 +102,33 @@ public ActionRequestValidationException validate() {
      */
     @Override
     public PutMappingRequest indices(String... indices) {
+        if (concretIndex != null) {
+            throw new IllegalStateException("indices can only be set if no concrete index is set");
+        }
```

Comment:
maybe put these checks into validate() method? The checks are then all in one place (the disadvantage is that exception happens later).